File: frontend/src/types/ride.ts
export interface Ride {
  id: string;
  rider_id: string;
  status: string;
  pickup_lat: number;
  pickup_lng: number;
  pickup_address?: string;
  dest_lat: number;
  dest_lng: number;
  dest_address?: string;
  vehicle_type: string;
  payment_method: string;
  surge_multiplier: number;
  estimated_fare?: number;
  matched_driver_id?: string;
  trip_id?: string;
  total_fare?: number;
  created_at: string;
}

File: backend/app/schemas/ride.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class RideRequestCreate(BaseModel):
    rider_id: UUID
    pickup_lat: float
    pickup_lng: float
    pickup_address: str | None = None
    dest_lat: float
    dest_lng: float
    dest_address: str | None = None
    vehicle_type: Literal["auto", "mini", "sedan", "suv"]
    payment_method: Literal["cash", "card", "wallet"] = "cash"


class RideRequestResponse(BaseModel):
    id: UUID
    rider_id: UUID
    status: str
    pickup_lat: float
    pickup_lng: float
    pickup_address: str | None
    dest_lat: float
    dest_lng: float
    dest_address: str | None
    vehicle_type: str
    payment_method: str
    surge_multiplier: float
    estimated_fare: float | None
    matched_driver_id: UUID | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class RideOfferResponse(BaseModel):
    id: UUID
    ride_id: UUID
    driver_id: UUID
    status: str
    expires_at: datetime
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
